File: pset2-arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h> //strlng
#include <math.h> //round function

void count(string sentence);
float letters = 0;
float words = 0;
float sentences = 0;

int main(void)
{
    string sentence = get_string("Text: ");
    count(sentence);
    words = words + 1; //there is 1 more word in the input than
    float grade = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8; //Coleman-Liau index.
    int roundgrade = round(grade);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", roundgrade);
    }
}

void count(string sentence)
{
    for (int i = 0, n = strlen(sentence); i < n; i++) // loops throug evry char of the string
    {
        if (isupper(sentence[i]) || islower(sentence[i])) //if it is a letter
        {
            letters++;
        }
        else if (sentence[i] == ' ') //if it is space marking a new word
        {
            words++;
        }
        else if (sentence[i] == '!' || sentence[i] == '.' || sentence[i] == '?') // if it is ! ? . markin a new sentence
        {
            sentences++;
        }
    }
}

File: Week 6 - Python/sentimental-readability/readability.py
from cs50 import get_string

text = get_string('Text: ')

letters = 0
# Because words are detectet by ' ' there will be one more word then countet
words = 1
sentences = 0
for char in text:
    # Checks for alphabetic charecters
    if char.isalpha():
        letters += 1
    # Checks for new words
    elif char == ' ':
        words += 1
    # Checks for sentences
    elif char == '!' or char == '.' or char == '?':
        sentences += 1
# Coleman-Liau index
grade = round(0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8)

if grade < 1:
    print('Before Grade 1')
elif grade > 16:
    print('Grade 16+')
else:
    print(f'Grade {grade}')
